### ISSUE TITLE
Provide support for conditional delete methods with annotation

### ISSUE THREAD

**Issue by bala-striva** [javers/javers#1080]
**Is your feature request related to a problem? Please describe.**
When using spring boot `JpaRepository` with custom delete methods i would like to use a javers annotation similar to `JaversAuditableDelete`. 

E.g.:
```
@Repository
class DummyAuditedRepository {
        List<DummyObject> deleteByName(String name) { ... }
}
```

**Describe the solution you'd like**

* create a new annotation for example `JaversAuditableConditionalDelete`
* add AOP handler for the new annotation in `JaversAuditableAspect`
* add a new method in `JaversCommitAdvice`
** collect the returned object that it's a collection of entities
** handle entity removal similar to `commitDeleteMethodArguments` by calling `commitShallowDelete`

**Describe alternatives you've considered**
Handle the shallow delete calls directly from the caller of the repository without support from javers.

**Comment by bartoszwalacik** [comment]
@bala-striva thanks for this high quality contribution, I will merge today

**Comment by bartoszwalacik** [comment]
released in 6.1.0
